### DISCUSSION TITLE
Is there a way to parse html in getStaticProps?

### DISCUSSION THREAD

**Discussion by es-ki** [vercel/next.js#30684] [Help] [answered]
After some research, I found the following method.  
[javascript - How to use DOMParser in next.js? - Stack Overflow](https://stackoverflow.com/questions/67915232/how-to-use-domparser-in-next-js)

When I tried to use [node-html-parser](https://www.npmjs.com/package/node-html-parser) in ```getStaticProps```, I got the following error message.

```
Server Error
TypeError: (0 , node_html_parser__WEBPACK_IMPORTED_MODULE_18__.default) is not a function

const converter = new QuillDeltaToHtmlConverter(body, config);
const articleBody = converter.convert();
const html = parse(articleBody);
                  ▲
console.log(html);
```

Is there a way to parse html in ```getStaticProps```? 

---

The purpose is to give an id attribute to a specific html tag in order to create a toc (table of contens).

```typescript
const converter = new QuillDeltaToHtmlConverter(body);
const articleBody = converter.convert(); // "<div><h2>foo</h2><h3>bar<h3><p>baz</p></div>"
const html = parse(articleBody);
const headings = [...html.window.document.querySelectorAll("h2")];
headings.forEach((elm, index) => {
  elm.id = `chapter-${index}`;
})
...
```

**Comment by magicspon** [comment, marked as answer]
example here takes some html from a cms rich text fields... converts the images to next/image, then converts it all back into a string.

https://gist.github.com/magicspon/1095cd5267adda9329383a63d130fdf9

**Comment by es-ki** [comment]
@magicspon 
That solution has worked for me.
Thank you very much.
